ext/web/timers.rs

```diff
@@ -23,29 +24,50 @@ pub trait TimersPermission {
 
 pub type StartTime = Instant;
 
+static mut NOW_BUF: *mut u32 = std::ptr::null_mut();
+
+#[op]
+pub fn op_now_set_buf(buf: ZeroCopyBuf) {
+  assert_eq!(buf.len(), 8);
+  // SAFETY: This is safe because this is the only place where we initialize
+  // NOW_BUF.
+  unsafe {
+    NOW_BUF = buf.as_ptr() as *mut u32;
+  }
+}
+
 // Returns a milliseconds and nanoseconds subsec
 // since the start time of the deno runtime.
 // If the High precision flag is not set, the
 // nanoseconds are rounded on 2ms.
-#[op]
-pub fn op_now<TP>(state: &mut OpState) -> f64
+#[op(fast)]
+pub fn op_now<TP>(state: &mut OpState)
 where
   TP: TimersPermission + 'static,
 {
   let start_time = state.borrow::<StartTime>();
   let elapsed = start_time.elapsed();
   let seconds = elapsed.as_secs();
-  let mut subsec_nanos = elapsed.subsec_nanos() as f64;
-  let reduced_time_precision = 2_000_000.0; // 2ms in nanoseconds
+  let mut subsec_nanos = elapsed.subsec_nanos();
 
   // If the permission is not enabled
   // Round the nano result on 2 milliseconds
   // see: https://developer.mozilla.org/en-US/docs/Web/API/DOMHighResTimeStamp#Reduced_time_precision
   if !state.borrow_mut::<TP>().allow_hrtime() {
+    let reduced_time_precision = 2_000_000; // 2ms in nanoseconds
     subsec_nanos -= subsec_nanos % reduced_time_precision;
   }
 
-  (seconds * 1_000) as f64 + (subsec_nanos / 1_000_000.0)
+  // SAFETY: This is safe because we initialize NOW_BUF in op_now_set_buf, its a null pointer
+  // otherwise.
+  // op_now_set_buf guarantees that the buffer is 8 bytes long.
+  unsafe {
```

Comment:
is it possible to call `Deno.core.ops.op_now.fast()` from userland? if so, won't we segfault here?

Reply:
NOW_BUF is null if not set. `if !NOW_BUF.is_null() {` This check makes sure we don't dereference it.